src/Shared/JSInterop/JSCallResultTypeHelper.cs

```diff
@@ -19,6 +19,11 @@ public static JSCallResultType FromGeneric<TResult>()
             {
                 return JSCallResultType.JSObjectReference;
             }
+            else if (typeof(TResult).Assembly == _currentAssembly
+                && typeof(TResult) == typeof(IJSDataReference))
```

Comment:
nit: we put `&&` at the end of the previous line by convention.

Reply:
> we put && at the end of the previous line by convention

Really? Ugh. That makes it much harder to skim-read since in left-to-right languages we skim from the left-hand-side of each line, not the end.